tap_shopify/context.py

```diff
@@ -23,6 +23,24 @@ def is_selected(cls, stream_name):
         stream_metadata = metadata.to_map(stream['metadata'])
         return metadata.get(stream_metadata, (), 'selected')
 
+    @classmethod
+    def get_unselected_fields(cls, stream_name):
+        stream = cls.get_catalog_entry(stream_name)
+        stream_metadata = metadata.to_map(stream['metadata'])
+        # All fields defined in the schema
+        all_fields = set(stream["schema"]["properties"].keys())
+
+        # Selected fields from metadata
+        selected_fields = set()
+        for breadcrumb, data in stream_metadata.items():
+            if len(breadcrumb) != 2:
+                continue # Skip root and nested metadata
+
+            if data.get('selected') or data.get('inclusion') == 'automatic':
+                selected_fields.add(breadcrumb[1])
```

Comment:
i think its simple to read this way?
```suggestion
            if len(breadcrumb) == 2:
              if data.get('selected') or data.get('inclusion') == 'automatic':
                  selected_fields.add(breadcrumb[1])
            
```